indico/modules/events/contributions/controllers.py

```diff
@@ -82,15 +90,38 @@ def _checkParams(self, params):
         self.contrib = Contribution.find_one(id=request.view_args['contrib_id'], is_deleted=False)
 
 
+class RHManageSubContributionBase(RHManageContributionBase):
+    """Base RH for a specific subcontribution"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.subcontrib
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageContributionBase._checkParams(self, params)
+        self.subcontrib = SubContribution.get_one(request.view_args['subcontrib_id'], is_deleted=False)
+
+
 class RHManageContributionsActionsBase(RHManageContributionsBase):
-    """Base class for classes performing actions on registrations"""
+    """Base class for classes performing actions on event contributions"""
 
     def _checkParams(self, params):
         RHManageContributionsBase._checkParams(self, params)
-        ids = set(map(int, request.form.getlist('contribution_id')))
+        ids = {int(c_id) for c_id in request.form.getlist('contribution_id')}
         self.contribs = self.event_new.contributions.filter(Contribution.id.in_(ids), ~Contribution.is_deleted).all()
 
 
+class RHManageSubContributionsActionsBase(RHManageContributionBase):
+    """Base class for RHs performing actions on subcontributions"""
+
+    def _checkParams(self, params):
+        RHManageContributionBase._checkParams(self, params)
+        ids = {int(c_id) for c_id in request.form.getlist('subcontribution_id')}
```

Comment:
`c_id`? I guess that's why @ThiefMaster always prefers using a less descriptive (`x`) var name for comprehensions.